pkg/core/config/main.go

```diff
@@ -539,13 +539,13 @@ func (config *Config) validateTargets() error {
 			}
 		}
 
-		if t.DisableConditions && len(t.ConditionIDs) > 0 {
-			logrus.Errorf("target %q has 'disableconditions' set to true and 'conditionids' defined (%v), it's not possible to disable conditions and define conditions at the same time", id, t.ConditionIDs)
+		if t.DisableConditions && len(t.DeprecatedConditionIDs) > 0 {
```

Comment:
```suggestion
		if t.DisableConditions && len(t.DeprecatedConditionIDs) > 0 {
```

Why does it becomes deprecated?

Reply:
Re-reading the code, I now understand 

```
Deprecated - please use DependsOn with `condition#conditionid` keys
```

Reply:
Exactly. In my opinion, the existence of `ConditionIds` is because, prior to this PR, a target by default depends on all of its conditions with an implicit `AND`. You could modify this behavior by enabling the `DisableConditions` option and adding a few IDs to `ConditionIDs`.

This behavior has now been generalized with the cross-resource DependsOn, so that:
```yaml
targets:
  conditionids:
     disableconditions: true
     conditionids:
        - 1
        - 3
  dependson:
     disableconditions: true
     dependson:
        - condition#1
        - condition#3
```
are equivalent. This is why I marked `ConditionIDs` as deprecated. From an implementation point of view, it's not critical—I've already implemented the logic for `ConditionIDs`, so we could mark them as non-deprecated. However, from a documentation and explanation point of view, I think it's easier to avoid having multiple ways to express the same behavior.

Reply:
Thanks for the clarification, it totally makes sense and you took the right approach